asserts/pool.go

```diff
@@ -478,36 +622,72 @@ func (p *Pool) Add(a Assertion, grouping Grouping) (ok bool, err error) {
 }
 
 func (p *Pool) addToGrouping(a Assertion, grouping Grouping, deserializeGrouping func(string) (*internal.Grouping, error)) (ok bool, err error) {
-	uniq := a.Ref().Unique()
-	var u *unresolvedRec
-	var extrag *internal.Grouping
-	var unresolved map[string]*unresolvedRec
-	if u = p.unresolved[uniq]; u != nil {
-		unresolved = p.unresolved
-	} else if u = p.prerequisites[uniq]; u != nil {
-		unresolved = p.prerequisites
+	var uniq string
+	ref := a.Ref()
+	// deal with sequence key
+	if !ref.Type.SequenceForming() {
+		uniq = ref.Unique()
 	} else {
-		ok, err := p.isPredefined(a.Ref())
-		if err != nil {
-			return false, err
+		atseq := AtSequence{
+			Type:        ref.Type,
+			SequenceKey: ref.PrimaryKey[:len(ref.PrimaryKey)-1],
```

Comment:
Sequence-forming assertions have "sequence point" number as the last element of their primary key - yes. 

About "combining" into "sequence group" - not really. When we drop that number we're dealing with **sequence key**. For the purpose of tracking of unresolved or to-be-updated sequence-forming assertions inside pool we use 'unresolved' lookup whose keys are derived from **sequence keys** (uniq = atseq.Unique() below; sequence point dropped); this is because by design we are only dealing with a single (specific) sequence-forming assertion to update/resolve at a time - for example, we might be updating a sequence-forming assertion with primary key ["foo", "bar", 3], but we would never update ["foo", "bar", 2] and ["foo", "bar", 3] at the same time (it's actually prevented by the respective entry code in this PR). I hope this makes sense and answers your question?

Reply:
Yes this all makes sense, thanks for clarifying